pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -649,9 +649,9 @@ func (ctrl *Controller) syncContainerRuntimeConfig(key string) error {
 	return ctrl.syncStatusOnly(cfg, nil)
 }
 
-// cleanUpDuplicatedMC removes the MC of uncorrected version if format of its name contains 'generated-xxx'.
-// BZ 1955517: upgrade when there are more than one configs, these generated MC will be duplicated
-// by upgraded MC with number suffixed name (func getManagedKeyCtrCfg()) and fails the upgrade.
+// cleanUpDuplicatedMC removes the MC of non-updated GeneratedByControllerVersionKey if its name contains 'generated-containerruntimeconfig'.
+// BZ 1955517: upgrade when there are more than one configs, the duplicated and upgraded MC will be generated (func getManagedKubeletConfigKey())
+// MC with old GeneratedByControllerVersionKey fails the upgrade.
 func (ctrl *Controller) cleanUpDuplicatedMC() error {
```

Comment:
Hmm, so in 4.11, we never removed the containerruntime de-dupe function?

Reply:
Yes. the cleanup function was removed from kubeletconfig and containerruntimeconfig controller in separate PRs that merged in the different releases.